lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,137 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], fmt="%g", mode="edge",
+                  padding=None, shifting=0, autoscale=True, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        captions : array-like, optional
+            A list of label texts, that should be displayed. If not given, the
+            label texts will be the data values formatted with *fmt*.
+
+        fmt : str, optional
+            A format string for the label. Default is '%g'
+
+        mode : {'edge', 'center'}, optional, default: 'edge'
+            Position of the label relative to the bar:
+
+            - 'edge': Placed at edge, the cumulative values will be shown.
+            - 'center': Placed at center, the individual values will be shown.
+
+        padding : float, optional
+            Space in points for label to leave the edge of the bar.
+
+        shifting : float, optional
```

Comment:
This variable name feels not quite right. But I don't have a better either right now.

Reply:
displacement? :thinking:

Reply:
padding => longitudinal_offset
shifting => lateral_offset ? :thinking: